### PR TITLE
Update contribution guide

### LINKED ISSUE #84: Contribution guide improvements
This issue is to collect challenges and confusion that new team members face as they follow the steps in the contribution guide so that we can improve the document.

One major issue is that "Step 3: Write the Local Neighborhood wrapper functions" is a huge leap from the previous steps. More scaffolding or smaller sub-steps are needed to build toward full implementations of `generate_inputs`, `run_container`, and `parse_output`. For instance, that could involve interactive inspecting the data structures available through the `Dataset` class and running pandas queries with them before implementing `generate_inputs`.

### PR DESCRIPTION
**Pull request by agitter**
Closes #84

This first draft addresses most of the ideas we came up with in #84 but not yet everything. @Lyce24 @ellango2612 @livj4711 you all went through the contribution guide most recently, so you can please give feedback on these updates. Are there parts that require more detail or instructions?

The new Step 0 on forking the repository feels too complicated to me. I like the idea of providing guidance here, but there are many ways to do this.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -73,8 +111,33 @@ Call the new class within `local_neighborhood.py` `LocalNeighborhood` and set `_
...
+0           A           B    0.98
+1           B           C    0.77
+```
+Also test the functions available in the `Dataset` class.
+```python
+> data.request_node_columns(['sources'])
+  sources NODEID
+0    True      A
+```
+Note the behavior of the `request_node_columns` function when there are missing values in that column of the node table.
```
Comment (livj4711) [inline CONTRIBUTING.md:136]: Also make clear that the NODEID column is always returned by this function

**Comment by agitter** [comment]
The DOMINO pull request #103 adds a new node attribute "active" that will need to be considered when generating inputs for the local neighborhood algorithm.

**Comment by agitter** [comment]
@zhiqianx does the change in 08e2575 clarify how the run function should prepare the output file? If not, can you please make a suggestion.

**Comment by ctrlaltaf** [comment]
The new contributing guide looks better and clarifies many of the steps that was difficult.

I noticed now on step 3's parse output function it says `"See the add_rank_column function in src.util.py"` but I was unable to find this. Is this function in src/util.py or is this something that has not been merged yet?

@agitter This step is related to when someone tries to run an algorithm through spras using `snakemake --cores 1 --configfile config/config.yaml`. I noticed that suppose local neighborhood was able to produce all the inputs and outputs correctly in the newly generated `output` and `.snakemake` directory. If changes are made to the code that break the workflow, for example the `parse_output` function now does not read the csv file correctly, and the user has not deleted the previously generated `output` and `.snakemake` directory, then spras does not throw an error since I believe the correct output files are already there(?). I understand this is a niche case, but would it be wise to include something like this in step 4:

`After running the above spras command, if everything goes well, two new directories 'output' and '.snakemake' will be generated. The output directory verifies if the algorithm is working and is a good thing to investigate if you have errors. It may also be useful that once you have successfully implemented the algorithm and do not need the output files anymore, you delete using rm -rf output .snakemake `

**Comment by agitter** [comment]
Thanks for the feedback @ctrlaltaf

The `add_rank_column` will be added by https://github.com/Reed-CompBio/spras/pull/92. The new instructions anticipate that other pull request being merged soon, but it is confusing in the meantime.

I agree we need to give more guidance about what Snakemake does once the outputs have been written. You shouldn't even have to remove the `.snakemake` directory though. That is used to track history of past runs and logs. Do the updates in 4de05999ff10528981ede43692badff666cd9120 help?

**Comment by ctrlaltaf** [comment]
> Thanks for the feedback @ctrlaltaf
> 
> The `add_rank_column` will be added by #92. The new instructions anticipate that other pull request being merged soon, but it is confusing in the meantime.
> 
> I agree we need to give more guidance about what Snakemake does once the outputs have been written. You shouldn't even have to remove the `.snakemake` directory though. That is used to track history of past runs and logs. Do the updates in [4de0599](https://github.com/Reed-CompBio/spras/commit/4de05999ff10528981ede43692badff666cd9120) help?

Yes I think the new addition is helpful! Thank you for clarifying the .snakemake

**Pull request merged**